include/ginkgo/core/base/dim.hpp

```diff
@@ -152,7 +156,31 @@ struct dim {
         return dim(x.first_ * y.first_, x.rest_ * y.rest_);
     }
 
+    /**
+     * A stream operator overload for dim
+     *
+     * @param os  stream object
+     * @param x  dim object
+     *
+     * @return a stream object appended with the dim output
+     */
+    friend GKO_ATTRIBUTES std::ostream &operator<<(std::ostream &os,
```

Comment:
nit: this overload is not working on devices, therefore:
```suggestion
    friend std::ostream &operator<<(std::ostream &os,
```